core/src/replay_stage.rs

```diff
@@ -3111,6 +3206,18 @@ impl ReplayStage {
 
             assert_eq!(bank_slot, bank.slot());
             if bank.is_complete() {
+                if let Some(first_alpenglow_slot) = first_alpenglow_slot {
+                    if !*is_alpenglow_migration_complete && bank.slot() >= first_alpenglow_slot {
+                        info!(
+                            "initiating alpenglow migration from replaying bank {}",
+                            bank.slot()
+                        );
+                        Self::initiate_alpenglow_migration(
+                            poh_recorder,
+                            is_alpenglow_migration_complete,
+                        );
+                    }
```

Comment:
I see @bw-solana closed the other conversation, but nobody responded to my last comment: https://github.com/anza-xyz/agave/pull/7450#discussion_r2299660484

Not necessarily a hill I'll die on, but would like to hear a counter point I guess 😄. We could seemingly de-duplicate some logic by shifting more into a `maybe_initiate_alpenglow_migration()` function (ie, the `is_some()` and slot-greater-than-first_alpenglow_slot check)